crates/client/rpc/src/providers/forward_to_provider.rs

```diff
@@ -1,26 +1,27 @@
 use crate::{bail_internal_server_error, errors::StarknetRpcApiError};
 use jsonrpsee::core::{async_trait, RpcResult};
+use mc_gateway_client::GatewayProvider;
+use mp_gateway::error::SequencerError;
 use mp_transactions::BroadcastedDeclareTransactionV0;
 use starknet_core::types::{
     BroadcastedDeclareTransaction, BroadcastedDeployAccountTransaction, BroadcastedInvokeTransaction,
     DeclareTransactionResult, DeployAccountTransactionResult, InvokeTransactionResult,
 };
-use starknet_providers::{Provider, ProviderError};
 
 use super::AddTransactionProvider;
 
-pub struct ForwardToProvider<P: Provider + Send + Sync> {
-    provider: P,
+pub struct ForwardToProvider {
+    provider: GatewayProvider,
 }
 
-impl<P: Provider + Send + Sync> ForwardToProvider<P> {
-    pub fn new(provider: P) -> Self {
+impl ForwardToProvider {
+    pub fn new(provider: GatewayProvider) -> Self {
         Self { provider }
     }
 }
 
 #[async_trait]
-impl<P: Provider + Send + Sync> AddTransactionProvider for ForwardToProvider<P> {
+impl AddTransactionProvider for ForwardToProvider {
```

Comment:
note for later: i'd love to move AddTransactionProvider to its own crate

Reply:
I also asked myself the same question
I split the gateway crate into client and server (which is a good thing) due to cyclic dependencies caused by this 😅

Reply:
ah yes
i think of this addtxprovider trait as an abstraction from the frontend services and backend services in madara, and it would allow mc-eth (or other future l1 watchers) / mc-rpc / mc-gateway / mc-p2p not to have to depend on mc-rpc
i think if we do this we should redefine our own error enum for add tx provider, and implement the Into starknet types rpc error in mc-rpc for it (and in gateway, p2p too then)